File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/CommandProcessorFactory.java
package ro.razvan.java.dagger.atm;

import dagger.Component;

import javax.inject.Singleton;

@Component(modules = {
        LoginCommandModule.class,
        HelloWorldCommandModule.class,
        SystemOutModule.class,
        UserCommandsRouter.InstallationModule.class,
        AmountsModule.class
})
@Singleton
public interface CommandProcessorFactory {

    static CommandProcessorFactory create() {
        return DaggerCommandProcessorFactory.create();
    }

    CommandProcessor processor();
}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/CommandProcessorFactory.kt
package ro.razvan.kotlin.dagger.atm

import dagger.Component
import javax.inject.Singleton

@Component(
    modules = [
        UserCommandsRouter.InstallationModule::class,
        HelloWorldCommandModule::class,
        LoginCommandModule::class,
        SystemOutModule::class
    ]
)
@Singleton
interface CommandProcessorFactory {
    fun processor(): CommandProcessor

    companion object {
        fun create(): CommandProcessorFactory =
            DaggerCommandProcessorFactory.create()
    }
}